share/spice/rx_info/rx_info.js

```diff
@@ -19,40 +19,45 @@
         return concatted;
     }
 
+    // Use matchedTerms object to build secondaryText string
+    function buildSecondaryText(matchedTerms) {
+        var secondaryText = "";
+
+        // Loop through matchedTerms object for terms and values
+        for (var key in matchedTerms) {
+           if (matchedTerms.hasOwnProperty(key)) {
+              secondaryText += key + "=" + matchedTerms[key] + ", ";
+           }
+        }
+
+        // If terms were found, make secondaryText human readable
+        if (secondaryText.length > 0) {
+            secondaryText = "Matched: " + secondaryText.substr(0, secondaryText.length - 2) + ".";
+        }
+
+        return secondaryText;
+    }
+
     env.ddg_spice_rx_info = function(api_result){
 
         if (!api_result || api_result.error || !api_result.replyStatus || !api_result.replyStatus.totalImageCount || api_result.replyStatus.totalImageCount < 1) {
             return Spice.failed('rx_info');
         }
         
-        var script = $('[src*="/js/spice/rx_info/"]')[0],
-            source = $(script).attr("src"),
-            triggerWordMatch = parseInt(source.match(/rx_info\/[^\/]+\/(\d)/)[1]),
-            relCheck;
-        
+        var sourceName = "DailyMed",
+            sourceUrl = "http://dailymed.nlm.nih.gov/",
```

Comment:
Zaahir,

Thank you so much for the response. moreText is a great new feature! The
next release of our code will take advantage of it. I will keep you posted.

On Wed, Sep 30, 2015 at 3:51 PM, Zaahir Moolla <notifications@github.com>
wrote:

> In share/spice/rx_info/rx_info.js
> <https://github.com/duckduckgo/zeroclickinfo-spice/pull/1753#discussion_r40842394>
> :
>
> >      env.ddg_spice_rx_info = function(api_result){
> >
> >          if (!api_result || api_result.error || !api_result.replyStatus || !api_result.replyStatus.totalImageCount || api_result.replyStatus.totalImageCount < 1) {
> >              return Spice.failed('rx_info');
> >          }
> >
> > -        var script = $('[src*="/js/spice/rx_info/"]')[0],
> > -            source = $(script).attr("src"),
> > -            triggerWordMatch = parseInt(source.match(/rx_info\/[^\/]+\/(\d)/)[1]),
> > -            relCheck;
> > -
> > +        var sourceName = "DailyMed",
> > +            sourceUrl = "http://dailymed.nlm.nih.gov/",
>
> @medicosconsultants <https://github.com/medicosconsultants> sorry for the
> long delay!
>
> This is now possible using our new moreText property.
>
> You can have a "More at" link that points somewhere useful for the user,
> and use the moreText to mention the data source.
>
> The "Is It Raining?" Spice is a great example:
> https://github.com/duckduckgo/zeroclickinfo-spice/blob/master/share/spice/rain/rain.js#L33
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/duckduckgo/zeroclickinfo-spice/pull/1753/files#r40842394>
> .
>